datatable/java/datatable/src/test/java/io/cucumber/datatable/DataTableTypeRegistryTest.java

```diff
@@ -37,8 +46,132 @@ public Place transform(DataTable table) {
+
+        registerDefaultEntryTransformer();
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForCell());
+
+        assertNull(lookupTableTypeByType);
+    }
+
+    @Test
+    public void returns_null_data_table_type_for_entry_if_no_default_registered() throws Exception {
+
+        registerDefaultCellTransformer();
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForEntry());
+
+        assertNull(lookupTableTypeByType);
+    }
+
+
+
+    @SuppressWarnings({"unchecked", "ConstantConditions"})
+    @Test
+    public void returns_default_data_table_type_for_cell_if_none_match_and_default_registered() {
+
+        registerCellEntryAndDefaults();
+
+        String here="here";
+        DataTableType lookupTableTypeByTypeForCell = registry.lookupTableTypeByType(constructTypeForCell());
+        List<List<Place>> transformedCells = (List<List<Place>>) lookupTableTypeByTypeForCell.transform(singletonList(singletonList(here)));
+
+        assertEquals(1,transformedCells.size());
+        assertEquals(1,transformedCells.get(0).size());
+        assertEquals(here,transformedCells.get(0).get(0).name);
+    }
+
+    @SuppressWarnings({"unchecked", "ConstantConditions"})
+    @Test
+    public void returns_default_data_table_type_for_entry_if_none_match_and_default_registered() {
+
+        registerCellEntryAndDefaults();
+
+        DataTableType lookupTableTypeByTypeForEntry = registry.lookupTableTypeByType(aListOf(Place.class));
+
+        String here="here";
+        List<Place> transformedEntries = (List<Place>) lookupTableTypeByTypeForEntry.transform(Arrays.asList(Arrays.asList("name","index of place"), Arrays.asList(here,"20")));
+        assertEquals(1,transformedEntries.size());
+        assertEquals(here,transformedEntries.get(0).name);
+        assertEquals(20,transformedEntries.get(0).indexOfPlace);
+    }
+
+    @Test
+    public void returns_cell_data_table_type() {
+
+        DataTableType cell = DataTableType.cell(Place.class);
+        registry.defineDataTableType(cell);
+        registry.defineDataTableType(DataTableType.entry(Place.class));
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(constructTypeForCell());
+
+        assertSame(cell,lookupTableTypeByType);
+    }
+
+    @Test
+    public void returns_entry_data_table_type() {
+
+        registry.defineDataTableType(DataTableType.cell(Place.class));
+        DataTableType entry = DataTableType.entry(Place.class);
+        registry.defineDataTableType(entry);
+
+        DataTableType lookupTableTypeByType = registry.lookupTableTypeByType(aListOf(Place.class));
+
+        assertSame(entry, lookupTableTypeByType);
+    }
+
+    private JavaType constructTypeForCell() {
+        return aListOf(aListOf(Place.class));
+    }
+
+    private JavaType constructTypeForEntry() {
+        return aListOf(Place.class);
+    }
+
+    private void registerCellEntryAndDefaults()  {
```

Comment:
This method is hiding quite a bit of non-trivial test setup and it isn't immediately clear how the tests that use it are affected by it. Can you change the structure of the tests/code so that it is?